Exercise/src/my_string.cpp

```diff
@@ -18,49 +22,143 @@ char* Exercise_1::CMyString::TryToAllocate(const size_t nLength) noexcept {
+	char chCurrentChar = cszStringToCount[0];
+	while(chCurrentChar != '\0')
 	{
 		++nStringLength;
-		cCurrentChar = cszStringToCount[nStringLength];
+		chCurrentChar = cszStringToCount[nStringLength];
 	}
 	// Adding 1 because string is zero terminated
 	return nStringLength + 1;
 }
 
-void Exercise_1::CMyString::CopyString(const char *cszStringToCopy, char *szStringToPaste, std::size_t nCountOfChars)
+void Exercise_1::CMyString::CopyString(const char *cszStringToCopy)
 {
-	for(std::size_t nIndex = 0; nIndex < nCountOfChars; nIndex++)
+	if(cszStringToCopy == nullptr)
 	{
-		szStringToPaste[nIndex] = cszStringToCopy[nIndex];
+		return;
 	}
+
+	std::size_t nIndex = 0;
+	while(cszStringToCopy[nIndex] != '\0')
+	{
+		m_szData[nIndex] = cszStringToCopy[nIndex];
+		++nIndex;
+	}
+	m_szData[nIndex] = '\0';
 }
 
-Exercise_1::CMyString::CMyString(const char* cszCharsSequence) 
+void Exercise_1::CMyString::AppendToString(const char *cszStringToAppend)
 {
-	m_nSize = GetStringLength(cszCharsSequence);
-	m_szData = TryToAllocate(m_nSize);
+	if(cszStringToAppend == nullptr)
+	{
+		return;
+	}
+	
+	ReinitializeAndCopy(m_szData, size() + GetStringLength(cszStringToAppend) - 1);
+	
 
-	CopyString(cszCharsSequence, m_szData, m_nSize);
+	size_t nTerminantIndex = GetStringLength(m_szData) - 1;
+	for(size_t nIndex = nTerminantIndex; true; nIndex++)
+	{
+		const char cchCharToAppend = cszStringToAppend[nIndex - nTerminantIndex];
+		m_szData[nIndex] = cchCharToAppend;
+		if(cchCharToAppend == '\0')
+		{
+			break;
+		}
+	}
 }
 
-Exercise_1::CMyString::CMyString(const Exercise_1::CMyString& cStringToCopy)
+void Exercise_1::CMyString::Insert(const char* cszStringToInsert, const size_t nStartPosition, const size_t nCountOfCharsToInsert)
 {
-	// Cleaning data we used before
-	bool bShouldCleanup = m_szData != nullptr && m_nSize != cStringToCopy.m_nSize;
-	if(bShouldCleanup)
+	const size_t nLengthOfInserted = GetStringLength(cszStringToInsert);
+	if(nStartPosition > size() || nLengthOfInserted == 0)
 	{
-		delete[] m_szData;
+		printf("Tried to reach wrong position!\n");
+		return;
 	}
+	if(nLengthOfInserted < nCountOfCharsToInsert)
+	{
+		printf("Tried to copy more characters than present\n");
+		return;
+	}
+
+	const size_t nNewSize = size() + nCountOfCharsToInsert;
+	char *szNewString = new char[nNewSize]; 
+
+	size_t nCurrentPosition = 0;
+	// Copying first part
+	for(; nCurrentPosition < nStartPosition; nCurrentPosition++)
+	{
+		szNewString[nCurrentPosition] = m_szData[nCurrentPosition];
+	}
+	// Beggining multiple insertion of given strings 
+	size_t nEndPosition = nStartPosition + nCountOfCharsToInsert;
+	for(; nCurrentPosition < nEndPosition; nCurrentPosition++){
+		szNewString[nCurrentPosition] = cszStringToInsert[nCurrentPosition - nStartPosition];
+	}
+	// Appending last part
+	for(; nCurrentPosition < nNewSize; nCurrentPosition++)
+	{
+		szNewString[nCurrentPosition] = m_szData[nStartPosition + nCurrentPosition - nEndPosition];
+	}
```

Comment:
It is better to create 1 loop with more condition blocks.

Reply:
Fixed